torch_geometric/explain/algorithm/pgm_explainer.py

```diff
@@ -0,0 +1,462 @@
+import logging
+from typing import List, Optional, Tuple
+
+import numpy as np
+import torch
+
+from torch_geometric.explain.config import ModelTaskLevel
+from torch_geometric.explain.explanation import Explanation
+from torch_geometric.utils import k_hop_subgraph
+from torch_geometric.utils.subgraph import get_num_hops
+
+from .base import ExplainerAlgorithm
+
+
+class PGMExplainer(ExplainerAlgorithm):
+    r"""The PGMExplainer model from the `"PGMExplainer: Probabilistic
+    Graphical Model Explanations  for Graph Neural Networks"
+    <https://arxiv.org/abs/1903.03894>`_ paper
+
+    Args:
+        perturb_feature_list (List): indices of the perturbed features
+            for graph classification explanations
+        perturb_mode (str): which method to generate the variations in
+            features one of ['randint', 'mean', 'zero', 'max', 'uniform']
+        perturbations_is_positive_only (bool): whether to apply the
+            abs function to restrict perturbed values to be +ve
+        is_perturbation_scaled (bool): whether to normalise the range
+            of the perturbed features
+
+    The generated explanation will provide a node_mask where neighbouring
+    nodes that are statistically significant in the model providing
+    the prediction have a value of 1 and the others 0. There is also a
+    `pgm_stats` tensor which stores the p_values of each of the nodes
+    as calculated by the chi_square test (pgmpy.estimators.CITests.chi_square)
+    used to generate the node mask
+
+    """
+    def __init__(
+        self,
+        perturb_feature_list: List = None,
```

Comment:
Can we just perturb all features. From a users point of view what does changing this input mean?

Reply:
In the original repo for the paper, the authors have this option for the graph classifiers so only the selected features are perturbed and the rest remains the same-- I think it's because they use an aggregation function for the perturbation, you wouldn't want to apply it to all the features, otherwise everything will be the same for all the input samples? 

But happy to leave this out, or to have the default option to be the random perturbation on all the features

Reply:
Ah ok. I am not familiar with the entire paper. So i'd appreciate your help deciding. 
From a users perspective whats the difference between setting `perturb_features_list` to all features and a subset of features. How does a user interpret the results of the explainer for different `perturb_features_list`? If we can explain to the user how this effects explanations lets keep it.